extmod/machine_i2c.c

```diff
@@ -526,13 +526,24 @@ STATIC mp_obj_t machine_i2c_writevto(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(machine_i2c_writevto_obj, 3, 4, machine_i2c_writevto);
 
-STATIC int read_mem(mp_obj_t self_in, uint16_t addr, uint32_t memaddr, uint8_t addrsize, uint8_t *buf, size_t len) {
-    mp_obj_base_t *self = (mp_obj_base_t *)MP_OBJ_TO_PTR(self_in);
-    uint8_t memaddr_buf[4];
+STATIC size_t fill_memaddr_buf(uint8_t *memaddr_buf, uint32_t memaddr, uint8_t addrsize) {
     size_t memaddr_len = 0;
+    if (addrsize != 8 && addrsize != 16 && addrsize != 24 && addrsize != 32) {
```

Comment:
Just considering code size and the cost (in code bytes) of this change, it might make for smaller code to check that `addrsize & 7 == 0 && addrsize <= 32` (and addrsize of 0 would get through but not break anything).

Reply:
Good idea. That reduces text by 8 bytes. However it also increases data by 16 bytes for whatever reason. (ESP32)

```
if (addrsize != 8 && addrsize != 16 && addrsize != 24 && addrsize != 32)
   text    data     bss     dec     hex filename
1098964  279872   37364 1416200  159c08 build-GENERIC/application.elf
```

```
if ((addrsize & 7) != 0 || addrsize > 32)
   text    data     bss     dec     hex filename
1098956  279888   37364 1416208  159c10 build-GENERIC/application.elf
```

Reply:
The data increase ssems to have been an incremental build artifact:

```
if ((addrsize & 7) != 0 || addrsize > 32)
   text    data     bss     dec     hex filename
1098956  279872   37364 1416192  159c00 build-GENERIC/application.elf
```